src/core/logging/module-logger.ts

```diff
@@ -0,0 +1,76 @@
+import winston, { format, Logger } from 'winston';
+import { ConfigService } from '@nestjs/config';
+import { Inject } from '@nestjs/common';
+import util from 'util';
+import { ServerConfig } from '../../shared/config/index.js';
+import { LoggingConfig } from '../../shared/config/logging.config.js';
+
+const MODULE_NAME: string = 'MODULE_NAME';
+
+export const localFormatter: (info: winston.Logform.TransformableInfo) => string = (
+    info: winston.Logform.TransformableInfo,
+) => {
+    const message: string = typeof info.message === 'string' ? info.message : util.inspect(info.message);
+    let context: string = 'Nest';
+    let trace: string = '';
+    let timestamp: string = '';
+    let ms: string = '';
+
+    if (typeof info['context'] === 'string') {
+        context = info['context'];
+    }
+
+    if (typeof info['timestamp'] === 'string') {
+        timestamp = info['timestamp'];
+    }
+
+    if (typeof info['ms'] === 'string') {
+        ms = info['ms'];
+    }
+
+    if (typeof info['trace'] === 'string') {
+        trace = `\n    ${info['trace']}`;
+    }
+
+    return `${info.level}\t ${timestamp} (${ms})\t \x1b[33m[${context}]\x1b[39m - ${message}${trace}`;
+};
+
+export class ModuleLogger {
+    private logger: Logger;
+
+    private moduleNameInternal: string;
+
+    public constructor(@Inject(MODULE_NAME) moduleName: string, configService: ConfigService<ServerConfig, true>) {
+        this.moduleNameInternal = moduleName;
+
+        const loggerConfig: LoggingConfig = configService.getOrThrow<LoggingConfig>('LOGGING');
+        const configKey: keyof LoggingConfig = `${moduleName.toUpperCase()}_LOG_LEVEL` as keyof LoggingConfig;
```

Comment:
```suggestion
        const configKey: keyof LoggingConfig = `${moduleName.split(/(?=[A-Z])/).join('_').toUpperCase()}_LOG_LEVEL` as keyof LoggingConfig;
```
I think, this would do the trick to be still able to dynamically load config variables like 'PERSON_MODULE_LOG_LEVEL' despite the class name 'PersonModule'
The question is, if we want this. ;)

Reply:
It would make it better readable. will this work on 'PERSON_API_MODULE_LOG_LEVEL' as well?